File: ClientApp/ClientApp/Models/Requests/ResetPasswordRequest.cs
namespace ClientApp.Models.Requests
{
    
    class ResetPasswordRequest : IRequest
    {
        /// <summary>
        /// Initializes a new instance of the <see cref="ResetPasswordRequest"/> class.
        /// </summary>
        /// <param name="newPassword">The new password to set for the user.</param>
        /// <param name="email">The username of the user whose password is being reset.</param>
        public ResetPasswordRequest(string newPassword, string email, string tocken)
        {
            this.NewPassword = newPassword;
            this.Email = email;
            ResetPasswordTocken = tocken;
        }

        public RequestsCodes GetCode() => RequestsCodes.ResetPasswordRequest;

        public string Email { get; }

        public string NewPassword { get; }

        public string ResetPasswordTocken { get; }
    }
}

File: M_Trivia/M_Trivia/ResetPasswordRequest.cpp
#include "ResetPasswordRequest.hpp"

ResetPasswordRequest::ResetPasswordRequest(const nlohmann::json &j)
    : email(j.at("Email")), newPassword(j.at("NewPassword")),
      resetPasswordTocken(j.at("ResetPasswordTocken")) {}

const std::string &ResetPasswordRequest::getEmail() const {
  return email;
}

const std::string &ResetPasswordRequest::getNewPassword() const {
  return newPassword;
}

const std::string &ResetPasswordRequest::getResetPasswordTocken() const {
  return this->resetPasswordTocken;
}
